synapse/lib/crypto/passwd.py

```diff
@@ -0,0 +1,83 @@
+import os
+import hmac
+import hashlib
+import logging
+
+import synapse.exc as s_exc
+
+import synapse.lib.coro as s_coro
+
+from typing import AnyStr, Dict
+
+logger = logging.getLogger(__name__)
+
+PBKDF2 = 'pbkdf2'
+DEFAULT_PTYP = PBKDF2
```

Comment:
If we do want to make this a framework, this should be a default arg to get/checkShadowV2() so it can be specified by the caller on a per-use basis.